provisioning/plugins/corim-psa-decoder/classattributes.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/veraison/corim/comid"
+)
+
+type PSAClassAttributes struct {
+	ImplID []byte
+	Vendor string
+	Model  string
+}
+
+// extract mandatory ImplID and optional vendor & model
+func (o *PSAClassAttributes) FromEnvironment(e comid.Environment) error {
+	class := e.Class
+
+	if class == nil {
+		return fmt.Errorf("expecting class in environment")
+	}
+
+	classID := class.ClassID
+
+	if classID == nil {
+		return fmt.Errorf("expecting class-id in class")
+	}
+
+	implID, err := classID.GetImplID()
+	if err != nil {
+		return fmt.Errorf("could not extract implementation-id from class-id: %w", err)
+	}
+
+	o.ImplID, _ = implID.MarshalJSON()
```

Comment:
What is the need for MarshalJSON at line 34?

Reply:
The RPC gop encoder only knows about the default Go types, so we need to pack our custom data types into JSON-encoded `[]byte`s to avoid the RPC layer to mess with them.

Reply:
Is it then applicable to other elements as well, like Instance ID, Signer ID etc?

Reply:
sorry, scratch my previous comment (I though we were in a different part of the codebase)!

This MarshalJSON is just a convenient way to obtain a `string` from a `[]byte`.  If `comid.ImplID` had a `String()` method I'd used that.